docs/src/literate-tutorials/dg_heat_equation.jl

```diff
@@ -0,0 +1,411 @@
+# # [Discontinuous Galerkin Heat equation](@id tutorial-dg-heat-equation)
+#
+# ![](dg_heat_equation.png)
+#
+# *Figure 1*: Temperature field on the unit square with an internal uniform heat source
+# solved with inhomogeneous Dirichlet boundary conditions on the left and right boundaries and flux on the top and bottom boundaries.
+#
+#-
+#md # !!! tip
+#md #     This example is also available as a Jupyter notebook:
+#md #     [`dg_heat_equation.ipynb`](@__NBVIEWER_ROOT_URL__/examples/dg_heat_equation.ipynb).
+#-
+#
+# This example was developed
+# as part of the Google summer of code funded project
+# ["Discontinuous Galerkin Infrastructure For the finite element toolbox Ferrite.jl"](https://summerofcode.withgoogle.com/programs/2023/projects/SLGbRNI5)
+#
+# ## Introduction
+#
+# The heat equation is the "Hello, world!" equation of finite elements.
+# Here we solve the equation on a unit square with a uniform internal source, using the interior penalty formulation.
+# The strong form of the (linear) heat equation is given by
+#
+# ```math
+#  -\nabla \cdot (k \nabla u) = f  \quad \textbf{x} \in \Omega,
+# ```
+#
+# where $u$ is the unknown temperature field, $k$ the heat conductivity,
+# $f$ the heat source and $\Omega$ the domain. For simplicity we set $f = 1$
+# and $k = 1$. We will consider inhomogeneous Dirichlet boundary conditions such that
+# ```math
+# u(\textbf{x}) = 1 \quad \textbf{x} \in \partial \Omega_u^+, \\
+# u(\textbf{x}) = -1 \quad \textbf{x} \in \partial \Omega_u^-,
+# ```
```

Comment:
It took me some time to spot the difference in the subscript between u and n. To me, using capital N for Neumann, and capital D for Dirichlet would make it more visible and easy to read. Secondly, perhaps the information in the table below can be put in directly in the equations?

For example
```
\textbf{x} \in \partial \Omega_D^+ = \lbrace\textbf{x} : x_1 = 1.0\rbrace
```